src/LineageTree/lineageTree.py

```diff
@@ -2141,43 +2276,183 @@ def unordered_tree_edit_distance(
 
-            def norm(*args):
-                return 1
+            # def norm(x,adj1,y,adj2,delta):
+            #     return max(uted(x,adj1,[],[],delta = delta),
+            #                uted(y,adj2,[],[],delta = delta))
 
-        simple_tree_1, _ = self.get_simple_tree(n1,end_time=end_time)
-        simple_tree_2, _ = self.get_simple_tree(n2, end_time=end_time)
-        nodes1, adj1, corres1 = self.__edist_format(simple_tree_1)
-        nodes2, adj2, corres2 = self.__edist_format(simple_tree_2)
+        simple_tree_1, times1 = self.get_comp_tree(n1, end_time=end_time)
+        simple_tree_2, times2 = self.get_comp_tree(n2, end_time=end_time)
+        nodes1, adj1, corres1 = self._edist_format(simple_tree_1)
+        nodes2, adj2, corres2 = self._edist_format(simple_tree_2)
         if len(nodes1) == len(nodes2) == 0:
             return 0
         delta_tmp = partial(
-            delta, corres1=corres1, corres2=corres2, times=self.cycle_time
+            delta,
+            corres1=corres1,
+            corres2=corres2,
+            times1=times1,
+            times2=times2,
         )
-        return uted(nodes1, adj1, nodes2, adj2, delta=delta_tmp) / norm(
-            nodes1, nodes2
+        return uted(nodes1, adj1, nodes2, adj2, delta=delta_tmp) / norm(n1, n2)
+
+    def export_simple_nx_graph(self, start_time: int = None):
+        """
+        Creates a networkx tree graph. This function is to be used for visualization,
+        so only the start and the end of a branch are calculated, all cells in between are not taken into account.
+        Args:
+            start_time (int): From which timepoints are the graphs to be calculated.
+                                For example if start_time is 10, then all trees that begin
+                                on tp 10 or before are calculated.
+        returns:
+            G : list(networkx objects)
+            pos : list(dict(id:position))
+        """
+
+        mothers = list(self.time_nodes[0])
+        if start_time:
+            mothers = [
+                root for root in self.roots if self.time[root] <= start_time
+            ]
+        all_nodes = {}
+        all_edges = {}
+        for mom in mothers:
+            nodes = []
+            edges = []
+            cells = [mom]
+            while cells != []:
+                tmp_cells = []
+                for cell in cells:
+                    new_cells = self.successor.get(
+                        self.get_cycle(cell)[-1], []
+                    )
+                    nodes.append(self.get_cycle(cell)[-1])
+                    if cell != self.get_cycle(cell)[-1]:
+                        edges.append((cell, self.get_cycle(cell)[-1]))
+                    for new_cell in new_cells:
+                        nodes.append(new_cell)
+                        edges.append((self.get_cycle(cell)[-1], new_cell))
+                        tmp_cells.append(new_cell)
+                    cells = tmp_cells.copy()
```

Comment:
I'm not sure you want to do it that way, here is another shot:
```python
nodes = []
edges = []
to_do = [mom]
while to_do:
    current = to_do.pop(0)
    last_cell = self.get_cycle(current)[-1]
    next_cells = self[last_cell]
    if current != last_cell:
        nodes.extend([current, last_cell])
        edges.append((current, last_cell))
    edges.extend((last_cell, next_cell) for next_cell in next_cells)
    to_do.extend(next_cells)
```

Reply:
Are you ok with something like that?

        mothers = self.time_nodes[0]
        all_nodes = {}
        all_edges = {}
        for mom in mothers:
            edges = set()
            nodes = set()
            for branch in self.get_all_branches_of_node(mom):
                nodes.update((branch[0],branch[-1]))
                if len(branch)>1:
                    edges.add((branch[0],branch[-1]))
                succ = self.successor.get(branch[-1], [])
                for suc in succ:
                    edges.add((branch[-1], suc))
            all_edges[mom] = edges
            all_nodes[mom] = nodes
        for i,mother in enumerate(mothers):
                graph[i] = nx.DiGraph()
                graph[i].add_nodes_from(all_nodes[mother])
                graph[i].add_edges_from(all_edges[mother])